### ISSUE TITLE
This looks interesting...

### ISSUE THREAD

**Issue by Gitoffthelawn** [Noitidart/Screencastify#1]
...best wishes creating it! :+1:

**Comment by Noitidart** [comment]
Thanks very much for following and the motivation sir! :) I am extending the idea of NativeShot to record the screen. I got the name from "Browserify" npm module. :) It's also a reason for me to test out WebRTC and get familiar with it, find its limitations. And fill in the limitations with js-ctypes.

**Comment by Gitoffthelawn** [comment]
You're welcome!  Thanks for all your excellent tools!

Have you concluded that the only way to view/capture the area outside of FF's own window(s) is by using WebRTC?

**Comment by Noitidart** [comment]
In NativeShot I used js-ctypes. I am trying to make things as simple as possible, a lot of the heavy lifting was already done by WebRTC. It does capture outside of Firefox but it is limited, such as I can't highlight the cursor during movement,  I can't hide the cursor, I can't record system sound, its a bit tricky to get webcam to overlay the recorded screen right now. Still investigating. But for first release I'm thinking of just releasing it as fullscreen record and upload to youtube/twitter/imgur/gfycat etc.

**Comment by Gitoffthelawn** [comment]
Interesting.  If I got a vote (and I don't), I would say to avoid WebRTC if possible.  I tend to recommend keeping it disabled unless it is being actively used.  Plus, as you mentioned, the inability to record system sound is an issue.

But that's just my personal preference... whatever you make will be excellent.  I base that on your amazing accomplishments.

**Comment by Noitidart** [comment]
Your vote totally counts and I will do that! It is honestly my plan. I already started the platform work -

https://github.com/Noitidart/ostypes_playground/commits/audio-capture

That's for Windows. :) It will allow system capture, webcam video, mic, and screen as well as portions of screens.

However that platform work will take some time. Especially to get it ready for Mac and Linux. NativeShot took a couple months. So to start off with, I'll just release this limited API (webrtc) but feature it with uploading to various sources. Then add a nice GUI. Then use this work to pitch to others to join me in collaboration. I work on so many things, I need collaborators. But I need a ptich. So I am only going with webrtc just to pitch. Your vote is the smartest one and I totally agree and the end goal will be that! :) Other extensions like in Google Chrome that do this also use binaries because WebRTC is so limited.

**Comment by Gitoffthelawn** [comment]
Wow!  I'm going to frame your kind response!  Thank you!  :-)

BTW, there are a couple extensions that I use that could really use small improvements or rewrites.  They are almost trivial to code for someone with your knowledge and interests.  Nothing complex at all.  Would you be interested in playing with them?  If so, what's the best way to talk about them with you?

**Comment by Noitidart** [comment]
Thanks brother! :) Ideally if those extensions have Github repos could you please create an issue there and @ me. I personally also have addons like:
- Zoomr ([AMO](https://addons.mozilla.org/en-US/firefox/addon/zoomr/) | [Github](https://github.com/Noitidart/Zoomr/issues))
- MailtoWebmails (AMO | Github)

That need some major touch ups.

Then the ongoing ones that needs some minor touchups:
- NativeShot
- MouseControl beta - need to touch up to finally get a release out
- Profilist beta - need to touch up to finally get a release out

Some collaborations I'm trying to see out
- Sonfigier - (Github) ready for reelase with mic, waiting for sign off by collaborator. Then next version needs ability to record system audio
- FoxSpeak - (Github)

Then I have work and a personal promise to a teacher of mine to complete.

So I'm balancing a lot. But I would love to know what needs work though because I am actively trying to attract contributors to do especially the ones that need some easy touch ups. :)

**Comment by Gitoffthelawn** [comment]
That's quite a few projects!  :-)

Here's one that needs a little touchup: [_Double Click Closes Tab_](https://addons.mozilla.org/firefox/addon/double-click-closes-tab/)

It hasn't been updated in over 4 years, but it still works (almost) great.

The only issue I have with it is that, according to _about:performance_, it negatively impacts Firefox performance.  It's a _very_ simple extension, so it shouldn't have this issue.

The project isn't on GitHub (or anything similar), but the source is an easy read.

I'll save the other projects for later... I don't want to overwhelm you!

**Comment by Noitidart** [comment]
Haha thanks. Ideally I would like to get in touch with the author and release an update to their existing addon. Rather then creating another addon in parallel. For those ones you want touchups to can you try to get in touch with the author and get them set up to accept PRs. You are right that addon looks quite easy.

**Comment by Gitoffthelawn** [comment]
Yes, I've been wanting to email the original dev.  Like you, just too many projects!

**Comment by Noitidart** [comment]
Haha I know how you feel brother :)

AMO should make their design such that it's easy for users who are developers to say hey "i'm a developer and would love to help maintain this".

**Comment by Gitoffthelawn** [comment]
> AMO should make their design such that it's easy for users who are developers to say hey "i'm a developer and would love to help maintain this".

I agree completely.  There are several great extensions that just need a little tweaking.

The _Quick Search Bar_ extension was a **_fantastic**_ extension.  I really miss it.  The developer pulled it from AMO, I think because he was frustrated by changes made by Mozilla.  (There is a _QuickSearchBar_ - one word - extension on AMO, but it is not related.) I would love to see someone revive _Quick Search Bar_.

**Comment by Noitidart** [comment]
Hi there sir! I just want to say thank you so much for starring my amazon assist repo. I seriously seriously appreciate how you support me morally! I finally got screencastify approved. But I worked on it in June and by now things changed so fast. It works fairly well for "Monitor" mode though, please try it out - https://addons.mozilla.org/en-US/firefox/addon/screencastify/

I am now trying to do a non webRTC version as you requested, but got busy with work. :(

**Comment by Gitoffthelawn** [comment]
I'm so touched by your kind note, that I'm not even sure how to respond.  All I can say is you're welcome, and thank ***you***!  😃
